### PR TITLE
Fix disappearing tooltips with clickable elements inside

### LINKED ISSUE #2237: Cannot click on password change link
screen shot 2016-02-23 at 18 06 41
The link to the form for changing password cannot be clicked. One the mouse moves, the tip box disappears.

### PR DESCRIPTION
**Pull request by 4i60r**
Fixes #2237

### CODE REVIEW

File: src/ralph/admin/templates/admin/includes/javascripts.html [hunk 1]
```diff
@@ -97,6 +97,110 @@
...
+
+        Foundation.libs.tooltip.addListener('hide', function(target) {
+            var self = this;
+            var clickableTags = [
+                "a",
+                "button"
+            ];
+            var allowEvent = true;
+            var $tip = self.getTip(target);
+            clickableTags.forEach(function(tag) {
```
Comment (ar4s) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: IMO delay can works for all tooltips (with or without `clickableTags`)

File: src/ralph/admin/templates/admin/includes/javascripts.html [hunk 2]
```diff
@@ -97,19 +97,115 @@
...
+                var self = this;
+                if(self.listeners[evtName] === undefined) {
+                    self.listeners[evtName] = [];
+                }
+                self.listeners[evtName].push(listenerFunc);
+            },
+            triggerListeners: function(evtName, args) {
+                var self = this;
+                var cancelEvent = false;
+                if(self.listeners[evtName] !== undefined) {
```
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: ifSPACE(

File: src/ralph/admin/templates/admin/includes/javascripts.html [hunk 3]
```diff
@@ -97,19 +97,115 @@
...
+                    var listeners = self.listeners[evtName];
+                    var i;
+                    for(i = 0; i < listeners.length; i += 1) {
```
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: forSPACE(

File: src/ralph/admin/templates/admin/includes/javascripts.html [hunk 4]
```diff
@@ -97,19 +97,115 @@
...
+                'show': {
+                    old: null
+                },
+                'hide': {
+                    old: null
+                }
+            },
+            addListener: function(evtName, listenerFunc) {
+                var self = this;
+                if(self.listeners[evtName] === undefined) {
```
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: ifSPACE

File: src/ralph/admin/templates/admin/includes/javascripts.html (hunk 2 above)
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: typeof(var) !== "undefined"

File: src/ralph/admin/templates/admin/includes/javascripts.html (hunk 3 above)
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: http://javascript.crockford.com/code.html

File: src/ralph/admin/templates/admin/includes/javascripts.html [hunk 5]
```diff
@@ -97,19 +97,115 @@
...
+                }
+                return cancelEvent;
+            },
+            initEventSystem: function() {
+                var self = this;
+                Object.keys(self.proxyFunctions).forEach(function(funcName) {
+                    self.proxyFunctions[funcName].old = self[funcName];
+                    self[funcName] = function() {
+                        var cancelEvent = self.triggerListeners(funcName, arguments);
+                        /*
```
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: // if event is not canceled

File: src/ralph/admin/templates/admin/includes/javascripts.html [hunk 6]
```diff
@@ -97,19 +97,115 @@
...
+                            If the event is not cancelled, call the original function
+                        */
+                        if(!cancelEvent) {
```
Comment (vi4m) [inline src/ralph/admin/templates/admin/includes/javascripts.html]: ifSPACE

**Comment by mkurek** [comment]
we'll move the link outside the tooltip and won't show links in tooltips in the future

**Pull request closed without merging**